csrc/host_ir/jit.cpp

```diff
@@ -444,9 +448,29 @@ void unpackInputs(
 
   insertNvtxRangePush("unpackInputs", builder);
 
-  // Get the current function (main) and its first argument
   llvm::Function* func = builder.GetInsertBlock()->getParent();
-  llvm::Value* aten_tensor_array = func->getArg(0);
+
+  // Get the cacheId from the main function's first argument
+  llvm::Value* cache_id = func->getArg(0);
+  // NOTE: Currently we can only grab cacheId by traversing all vals
+  // In the future we should add a cacheId to the host ir container or fusion
+  bool found_cache_id = false;
+  Val* cache_id_val = nullptr;
+  for (Val* val : container->deterministic_vals()) {
+    if (auto* named_scalar = dynamic_cast<NamedScalar*>(val)) {
+      if (named_scalar->name() == "cacheId") {
+        if (found_cache_id) {
+          NVF_ERROR(named_scalar != cache_id_val, "cacheId is not the first deterministic val");
+        }
+        cache_id_val = named_scalar;
+        found_cache_id = true;
+      }
+    }
+  }
+  val_to_value[cache_id_val] = cache_id;
```

Comment:
should we move this to line 466?

Otherwise, we should guard it with `cache_id_val != nullptr`.